### PR TITLE
RootTask closes itself if all child tasks will be closed after_save

### LINKED ISSUE #13133: Appeals with post-dispatch tasks remain open
## Description

When post-dispatch tasks are opened on an Appeal (e.g. Mail tasks, or `InformalHearingPresentationTask`), the Root Task may remain on hold indefinitely and the Appeal appears "stuck" with no definite status. This is because [the status of the Appeal is derived from the state of its task tree](https://github.com/department-of-veterans-affairs/caseflow/issues/12713#issuecomment-554015297)

## Background/context/resources

#12238

The `AppealsWithNoTasksOrAllTasksOnHoldQuery` job reveals about 30 of these are consistently open.

## Technical notes

The existing mechanism for `update_parent_status` should be used to close the Root task where necessary. One way to do this would be to define a `Task.post_dispatch_task?` boolean method on the superclass and override it where necessary in the relevant child task classes. Then the `RootTask.when_child_task_completed` method could interrogate each child to see if the RootTask should close itself.

### PR DESCRIPTION
**Pull request by pkarman**
Resolves #13133 

### Description
Use existing callbacks to allow RootTask to close itself when there are only auto-close-able child tasks remaining.

This applies primarily to tasks opened *after* a case has been dispatched.

Tested in production, the `.post_dispatch_task?` correctly identifies 28 of the 30 stuck appeals.
```ruby
stuck_appeals.select { |appeal| appeal.tasks.any? { |t| t.open? && t.post_dispatch_task? } }
```

### CODE REVIEW

File: app/models/tasks/root_task.rb [hunk 1]
```diff
@@ -35,7 +43,14 @@ def when_child_task_created(child_task)
   end
 
   def update_children_status_after_closed
-    children.open.where(type: TrackVeteranTask.name).update_all(status: Constants.TASK_STATUSES.completed)
+    transaction do
+      # do not use update_all since that will avoid callbacks and we want versions history.
+      children.open.where(type: CHILD_TASK_TYPES_TO_CLOSE_AFTER_CLOSED).to_a.each(&:completed!)
```
Comment (hschallhorn) [inline app/models/tasks/root_task.rb]: Don't need to convert to array here.

Comment (pkarman) [reply app/models/tasks/root_task.rb]: I thought so too, but codeclimate would prefer I use `find_each` instead and that isn't what we want.

Comment (hschallhorn) [reply app/models/tasks/root_task.rb]: 😾 damn code climate

Comment (pkarman) [reply app/models/tasks/root_task.rb]: I mean, I guess we could use `find_each` but it seemed less intuitive to me. happy to switch if you'd prefer.

Comment (pkarman) [reply app/models/tasks/root_task.rb]: I changed it.

**Pull request merged**